crypto/property/property.c

```diff
@@ -353,11 +402,19 @@ int ossl_method_store_add(OSSL_METHOD_STORE *store, const OSSL_PROVIDER *prov,
 
         if (tmpimpl->provider == impl->provider
             && tmpimpl->properties == impl->properties)
-            break;
+            goto err;
```

Comment:
This is not correct. It will cause `alg_cleanup` to be called that will free the alg. But it is still in use.